lib/trino-hive-formats/src/test/java/io/trino/hive/formats/avro/TestAirCompressorAsAvroCodec.java

```diff
@@ -0,0 +1,240 @@
+            case SNAPPY -> in -> {
+                CRC32 crc32 = new CRC32();
+                try {
+                    int offset = in.arrayOffset() + in.position();
+                    ByteBuffer out = ByteBuffer.allocate(Snappy.uncompressedLength(in.array(), offset, in.remaining() - 4));
+                    int size = Snappy.uncompress(in.array(), offset, in.remaining() - 4, out.array(), 0);
+                    ((Buffer) out).limit(size);
+
+                    crc32.reset();
+                    crc32.update(out.array(), 0, size);
+                    if (in.getInt(((Buffer) in).limit() - 4) != (int) crc32.getValue()) {
+                        throw new IOException("Checksum failure");
+                    }
+
+                    return out;
+                }
+                catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            };
+            case ZSTANDARD -> in -> {
+                try {
+                    NonCopyingByteArrayOutputStream baos = new NonCopyingByteArrayOutputStream(DEFAULT_BUFFER_SIZE);
+                    InputStream bytesIn = new ByteArrayInputStream(in.array(), in.arrayOffset() + in.position(),
+                            in.remaining());
+                    try (InputStream ios = ZstandardLoader.input(bytesIn, false)) {
+                        IOUtils.copy(ios, baos);
+                    }
+                    return baos.asByteBuffer();
+                }
+                catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            };
+        };
+    }
+
+    static Function<ByteBuffer, ByteBuffer> getTrinoAvroCompression(AvroCompressionKind compressionKind)
+    {
+        return switch (compressionKind) {
+            case NULL, DEFLATE -> Function.identity(); //Not tested because avro code reused for this compression kind
+            case SNAPPY -> in -> {
+                try {
+                    return SNAPPY_CODEC_SUPPLIER.get().compress(in);
+                }
+                catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            };
+            case ZSTANDARD -> in -> {
+                try {
+                    return ZSTD_CODEC_SUPPLIER.get().compress(in);
+                }
+                catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            };
+        };
+    }
+
+    static Function<ByteBuffer, ByteBuffer> getTrinoAvroDecompression(AvroCompressionKind compressionKind)
+    {
+        return switch (compressionKind) {
+            case NULL, DEFLATE -> Function.identity(); //Not tested because avro code reused for this compression kind
+            case SNAPPY -> in -> {
+                try {
+                    return SNAPPY_CODEC_SUPPLIER.get().decompress(in);
+                }
+                catch (IOException e) {
```

Comment:
Make the method `throws IOException` instead

Reply:
This is inside of the `Function<ByteBuffer, ByteBuffer>`.  Is there a checked exception Function Interface? (Ala JDBI's HandleConsumer<ExceptionClass>)

Reply:
Ah, I see this is returning a lambda

Reply:
deleted :)